src/tabmat/formula.py

```diff
@@ -0,0 +1,313 @@
+            scoped_cols = OrderedDict()
+            for scoped_term in scoped_terms:
+                if not scoped_term.factors:
+                    scoped_cols[
+                        "Intercept"
+                    ] = scoped_term.scale * self._encode_constant(
+                        1, None, {}, spec, drop_rows
+                    )
+                else:
+                    scoped_cols.update(
+                        self._get_columns_for_term(
+                            [
+                                self._encode_evaled_factor(
+                                    scoped_factor.factor,
+                                    spec,
+                                    drop_rows,
+                                    reduced_rank=scoped_factor.reduced,
+                                )
+                                for scoped_factor in scoped_term.factors
+                            ],
+                            spec=spec,
+                            scale=scoped_term.scale,
+                        )
+                    )
+            cols.append((term, scoped_terms, scoped_cols))
+
+        # Step 3: Populate remaining model spec fields
+        if spec.structure:
+            cols = self._enforce_structure(cols, spec, drop_rows)
+        else:
+            # for term, scoped_terms, columns in spec.structure:
+            # expanded_columns = list(itertools.chain(colname_dict[col] for col in columns))
+            # expanded_structure.append(
+            #     EncodedTermStructure(term, scoped_terms, expanded_columns)
+            # )
+
+            spec = spec.update(
+                structure=[
+                    EncodedTermStructure(
+                        term,
+                        [st.copy(without_values=True) for st in scoped_terms],
+                        # This is the only line that is different from the original:
+                        list(
+                            itertools.chain(
+                                *(
+                                    mat.get_names(col)
+                                    for col, mat in scoped_cols.items()
+                                )
+                            )
+                        ),
+                    )
+                    for term, scoped_terms, scoped_cols in cols
+                ],
+            )
+
+        # Step 4: Collate factors into one ModelMatrix
+        return ModelMatrix(
+            self._combine_columns(
+                [
+                    (name, values)
+                    for term, scoped_terms, scoped_cols in cols
+                    for name, values in scoped_cols.items()
+                ],
+                spec=spec,
+                drop_rows=drop_rows,
+            ),
+            spec=spec,
+        )
+
+
+class InteractableDenseMatrix(DenseMatrix):
+    def __mul__(self, other):
+        if isinstance(other, (InteractableDenseMatrix, int, float)):
```

Comment:
How does this square with `DenseMatrix.multiply()` expecting only a vector as input?